client/src/components/RoomDirectoryPanel.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { FirebaseContext } from '../firebaseContext'
+import { IChatRoom } from '../types'
+import { makeStyles } from '@material-ui/core/styles'
+import Button from '@material-ui/core/Button';
+import { IconButton } from '@material-ui/core';
+import { Cancel } from '@material-ui/icons';
+import InputBase from '@material-ui/core/InputBase';
+import SearchIcon from '@material-ui/icons/Search';
+import * as JsSearch from 'js-search';
+import HomeIcon from '@material-ui/icons/Home';
+
+const useStyles = makeStyles({
+    title: {
+        fontSize: 50,
+        fontWeight: 600,
+        marginBottom: 20,
+    },
+    container: {
+		width: 'fit-content',
+		maxWidth: window.innerWidth * 0.8,
+		borderRadius: 20,
+		borderWidth: 5,
+		display: 'flex',
+		justifyContent: 'center',
+		alignItems: 'center',
+		padding: 100,
+		flexDirection: 'column',
+		'& > *': {
+			fontFamily: 'Blinker'
+		},
+		backgroundColor: 'whitesmoke',
+		border: '5px solid #87D3F3',
+		position: 'relative'
+    },
+    cancelButton: {
+		position: 'absolute',
+		right: 10,
+		top: 10
+    },
+})
+
+const buttonStyles = makeStyles((theme) => ({
+    button: {
+      margin: theme.spacing(1),
+    },
+  }));
+
+
+interface IRoomDirectoryProps {
+    sendRoomDirectory: (roomDirectory: string, roomName: string) => void;
+}
+
+interface IEnterRoomProps {
+    roomName: string;
+    onClickCancel: () => void;
+}
+
+// interface IRoomIconProps {
+//     room: string;
+// }
+
+const pickNItemsArray = ( arr : Array<Object> | null, n : number) : Array<Object> | null => {
```

Comment:
This function is to randomly pick N items from the array. The reason for the function is I want to pick 10 random rooms from all the rooms. Random picking is to avoid the bias of always displaying the first 10 rooms every time.

Reply:
since we're already using the `underscore` library, you can use its shuffle function.
reference this code that similarly shuffles an array
https://github.com/LeoTheG/audio-editor/blob/master/client/src/routes/UploadPage.tsx#L309